src/complexityRule.ts

```diff
@@ -0,0 +1,647 @@
+                this.conditions.push(textChunk);
+            } else {
+                // Remove all the text that should be ignored: the contents of function calls and strings
+                // Otherwise the split() could match operators in those strings
+                const retVal = Utils.removeIgnoredText(textChunk);
+                textChunk = retVal[0];
+                this.ignoredText = retVal[1];
+
+                this.conditionChunks = textChunk.split(this.matchAndOrXor);
+
+                this.conditionChunks.forEach((condition) => {
+                    // Determine if an AND operator or an OR operator was found.
+                    // If so, store which was found and then remove it.
+                    if ((this.leadingAndMatch = condition.match(this.captureLeadingAnd)) !== null) {
+                        this.operators.push(new Operator(COND_TYPE.TYPE_AND));
+                        condition = condition.substring(this.leadingAndMatch[0].length);
+                    } else if ((this.leadingOrMatch = condition.match(this.captureLeadingOr)) !== null) {
+                        this.operators.push(new Operator(COND_TYPE.TYPE_OR));
+                        condition = condition.substring(this.leadingOrMatch[0].length);
+                    } else if ((this.leadingXorMatch = condition.match(this.captureLeadingXor)) !== null) {
+                        this.operators.push(new Operator(COND_TYPE.TYPE_XOR));
+                        condition = condition.substring(this.leadingXorMatch[0].length);
+                    }
+
+                    // Store anything that's not still empty.
+                    condition = condition.trim();
+                    if (condition !== '') {
+                        // Restore any text that was ignored above
+                        condition = Utils.restoreIgnoredText(condition, this.ignoredText);
+                        this.conditions.push(new Condition(condition));
+                    }
+                });
+
+            }
+        });
+
+
+        this.hasMixedOperators = Utils.hasMixedOperators(this.operators);
+
+        this.truePaths = this.hasMixedOperators ? null : new EvalTree(this, true);
+        this.falsePaths = this.hasMixedOperators ? null : new EvalTree(this, false);
+
+
+    }
+
+    getEvalPaths(treeType) {
+        return treeType ? this.truePaths : this.falsePaths;
+    };
+
+    hasMixedOperatorsDeep() {
+
+        return this.hasMixedOperators || this.conditions.some(function (c) {
+                return (c instanceof Expression) && c.hasMixedOperatorsDeep();
+            });
+    };
+}
+
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: 'complexity',
+        type: 'functionality',
+        description: 'Ensure that the two-way data binding syntax is correct.',
+        rationale: 'The parens "()" should have been inside the brackets "[]".',
```

Comment:
I guess both description and rationale are a WIP? :)

Reply:
Yes, a very early WIP ;)